website/src/controllers/groups.js

```diff
@@ -137,7 +137,25 @@ api.get = function(req, res, next) {
   q.exec(function(err, group){
     if (err) return next(err);
     if (!group && gid!=='party') return res.json(404,{err: "Group not found or you don't have access."});
-    res.json(group);
+
+    //Since we have a limit on how many members are populate to the group, we want to make sure the user is always in the group
+    var userInGroup = _.find(group.members, function(member){ return member._id == user._id; });
+
+    //If the group is private or the group is a party, then the user must be a member of the group based on access restrictions above
+    if (group.privacy === 'private' || gid === 'party') {
+      //If the user is not in the group query, add them
+      if (userInGroup === undefined) { group.members.push(user); }
```

Comment:
Lodash will return `undefined` in this case if the user is not found so in this case it's correct. The ! (not) operator will transform to `true` the following values: `null`, `undefined`, `0` `''` (empty string), `false` and `Nan` and when comparing these values they'll return true when using the `==` operator (`null == undefined` but `null !== undefined`). Here there's a more in depth explanation http://www.sitepoint.com/javascript-truthy-falsy/